platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/ValidationFinishedContext.java

```diff
@@ -35,7 +36,7 @@ protected ValidationFinishedContext(ValidationFinishedContext parent) {
     /**
      * Returns the list of validation warnings encountered so far.
      */
-    public ImmutableList<? extends Problem> getValidationProblems() {
+    public List<? extends Problem> getValidationProblems() {
```

Comment:
❓ Is it something EP complains about? `ImmutableList` has a stricter contract than a `List`, and the [recommendation from the guava authors](https://stackoverflow.com/a/9521003) is to be explicit about using Immutable types rather than hiding them behind JDK interfaces.

Reply:
No EP doesn't complain, but for quite a while now I think this is too much information on the interface, we only use very basic functionality of this list.